app/controllers/recipes_controller.rb

```diff
@@ -0,0 +1,80 @@
+class RecipesController < ApplicationController
+  before_action :set_recipe, only: %i[show edit update destroy]
+
+  # GET /recipes or /recipes.json
+  def index
+    @user = current_user
+    @recipes = current_user.recipes.includes(:user).order(created_at: :desc)
+  end
+
+  # GET /recipes/1 or /recipes/1.json
+  def show
+    @recipe_foods = @recipe.recipe_foods
+  end
+
+  # GET /recipes/new
+  def new
+    @recipe = Recipe.new
+  end
+
+  def public
+    @public_recipes = Recipe.includes([:user], [:recipe_foods]).where(public: true).order(created_at: :desc)
+    @public_recipes.each do |recipe|
+      recipe.recipe_foods.all.includes([:food]).sort_by { |_recipe_foods| recipe_food.food.name }
+    end
+  end
+
+  # GET /recipes/1/edit
+  def edit; end
+
+  # POST /recipes or /recipes.json
+  def create
+    @recipe = Recipe.new(recipe_params)
+    @recipe.user_id = params[:user_id]
+
+    respond_to do |format|
+      if @recipe.save
+        format.html { redirect_to user_recipes_path(@recipe), notice: 'Recipe was successfully created. ✅' }
+        format.json { render :show, status: :created, location: @recipe }
+      else
+        format.html { render :new, status: :unprocessable_entity }
+        format.json { render json: @recipe.errors, status: :unprocessable_entity }
+      end
+    end
+  end

```

Comment:
- [ ] When trying to create a recipe I get this error. The `cooking_time` seems to be properly permitted but does not appear in the record creation, so it is raising validation issues. Kindly fix that.

  ![jgue](https://user-images.githubusercontent.com/54848363/202840728-beeaaa75-a968-4f77-bd2e-b4e3f848daa7.jpg)